parley/src/shape.rs

```diff
@@ -38,9 +38,12 @@ pub fn shape_text<'a, B: Brush>(
     infos: &[(CharInfo, u16)],
     levels: &[u8],
     scx: &mut ShapeContext,
-    text: &str,
+    mut text: &str,
     layout: &mut Layout<B>,
 ) {
+    if text.is_empty() {
+        text = " ";
+    }
     // Do nothing if there is no text or styles (there should always be a default style)
     if text.is_empty() || styles.is_empty() {
```

Comment:
This first condition can now never be true, I think...

Reply:
We should probably leave the check there anyway, seeing as the hack to make text never empty is intended to be temporary?

Reply:
It’s temporary but I don’t expect to fix this hack until we redo shaping at which point this entire module will be rewritten so 🤷🏻‍♂️. 

Looking again, this should probably avoid setting empty text to a space if there are any inline objects.